py/configu/core/command.py

```diff
@@ -7,7 +7,7 @@
 
 
 @dataclass
-class Command(Generic[CommandReturn]):
+class Command(ABC, Generic[CommandReturn]):
```

Comment:
I've been wondering about this for a while...it might be better to use `Protocol[CommandReturn]` for duck-typing here.

My thinking is:
- We don't need to have `parameters` in the base class, allowing for parameter-less commands.
- We don't need to import subclass `Command` in various places
- Users wishing to implement their own commands don't have to subclass `Command`, their class just needs to implement a `run` method.

We can retain runtime-checking with the (runtime_checkable)[https://docs.python.org/3/library/typing.html#typing.runtime_checkable] decorator if we want.

Reply:
if we do decide to go down the  `Protocol` route, there's probably some additional refactoring work needed to achieve that. `ABC` is already a vast improvement, so would suggest this is fine for this PR and we look at converting it at a later date.

Reply:
I'd look into Protocol. for now let's keep it this way and as you suggested, PR Protocol.

Reply:
opened
https://github.com/configu/configu/issues/199#issue-1850979943